exporter/honeycombexporter/config_test.go

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package honeycombexporter
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/honeycombexporter/internal/metadata"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap/confmaptest"
+)
+
+func TestLoadConfig(t *testing.T) {
+	t.Parallel()
+
+	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
+	require.NoError(t, err)
+
+	//defaultCfg := CreateDefaultConfig()
+
+	tests := []struct {
+		id       component.ID
+		expected component.Config
+	}{
+
+		{
+			id: component.NewIDWithName("honeycomb", ""),
+			expected: &Config{
+				APIKey: "test-apikey",
+				APIURL: "https://api.testhost.io",
+				Markers: []marker{
+					{
+						MarkerType:   "fooType",
+						MessageField: "test message",
+						UrlField:     "https://api.testhost.io",
+						Rules: Rules{
+							ResourceConditions: []string{
+								`IsMatch(attributes["test"], ".*")`,
+							},
+							LogConditions: []string{
+								`body == "test"`,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			id: component.NewIDWithName("honeycomb", "color_no_type"),
+			expected: &Config{
+				APIKey: "test-apikey",
+				APIURL: "https://api.testhost.io",
+				Markers: []marker{
+					{
+						MarkerColor:  "green",
+						MessageField: "test message",
+						UrlField:     "https://api.testhost.io",
+						Rules: Rules{
+							ResourceConditions: []string{
+								`IsMatch(attributes["test"], ".*")`,
+							},
+							LogConditions: []string{
+								`body == "test"`,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "bad_syntax_log"),
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "no_conditions"),
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "no_api_key"),
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "no_api_url"),
```

Comment:
Lets update `CreateDefaultConfig()` to create a config with the prod url as the default url so that this test scenario isn't a failure.